apps/api/src/domain/games/memory-match/services/memory-match.service.ts

```diff
@@ -29,23 +33,38 @@ export class MemoryMatchService {
         }
 
         // our convention is to return not published when a user does not have access to a particular resource
-        if (!searchResult.isPublished) {
+        if (isNullOrUndefined(userWithGroups) && !searchResult.isPublished) {
+            return NotFound;
+        }
+
+        if (
+            !isNullOrUndefined(userWithGroups) &&
+            !userWithGroups.isAdmin() &&
+            !searchResult.isPublished
+        ) {
             return NotFound;
         }
 
         // at this point, we have a published round
         return this.buildView(searchResult);
     }
 
-    async fetchMany(): Promise<{ entities: IMemoryMatchRound[] }> {
+    async fetchMany(
+        userWithGroups?: CoscradUserWithGroups
+    ): Promise<{ entities: IMemoryMatchRound[] }> {
         // TODO filter out unpublished round at the level of the database
         const searchResult = await this.memoryMatchRepository.fetchMany();
 
+        const availableEntities = searchResult.filter((entity) => {
```

Comment:
Another possiblity is that you put a `canUser(user)` method on the `MemoryMatch` entity. Then you just check
```ts
if(entity.canUser(userWithGroups){
 // ...
}
```